### ISSUE TITLE
Settings page UI design for easier selection

### ISSUE THREAD

**Issue by jasiralavi** [Schneegans/Burn-My-Windows#229]
Hi, love your extension. Brings back memories of my old ubuntu desktop with compiz and all.
I think if you move the selections for various open and close animation to show inside the list of Effects (under Effect Options) it would make it easier to select unselect various effects. Please check my mockup below.

![burnmywindows ui redo](https://user-images.githubusercontent.com/16372030/199006621-c51b0f70-26f7-4a58-844f-0f049e45d532.png)

**Comment by Schneegans** [comment]
Hi there! Thanks for the suggestion, this could indeed be an improvement. I think I won't have the time to implement this in the near future, so if anybody wants to add this (it shouldn't be too hard), just drop a note here! When I start working on this, I'll self-assign the issue.

**Comment by Schneegans** [comment]
I used the opportunity to port more parts of the settings dialog to use libadwaita. So we now have basically three versions of the settings dialog:
1. On GNOME 3.36 and GNOME 3.38, the settings dialog is based on GTK3 and use a sidebar to switch to the effect pages.
2. On GNOME 40 and GNOME 41, the settings dialog is based on GTK4 and also uses the sidebar.
3. On GNOME 42 and GNOME 43, the settings dialog is based on GTK4 with libadwaita. **This has been redesigned with your proposal in mind.**

Each effect row now has three buttons: _"Preview the effect"_, _"Enable it for opening windows"_, and _"Enable it for closing windows"_. I opted for an icon-only approach. Maybe this takes a couple of seconds more to understand, but I think it's much more beautiful. All the buttons have tooltips as well.

![image](https://user-images.githubusercontent.com/829942/200397723-80d23000-8382-41ef-92f8-f62917ea2839.png)


If you click one of the effect rows, it is expanded. Hence, there are no sub-pages anymore. This requires much less clicking and mouse travel. I think this is a good solution. Thank you again for this idea!

![image](https://user-images.githubusercontent.com/829942/200397883-03730a10-a18b-4d89-86ec-d0e7ebb80927.png)
